src/Caliburn.Micro.Platform/XamlPlatformProvider.cs

```diff
@@ -106,9 +116,14 @@ public class XamlPlatformProvider : IPlatformProvider {
                         exception = ex;
                     }
                 };
+#if AVALONIA
+                //TODO: (Avalonia) Need to check if GetAwaiter().GetResult() is a good solution here
+                dispatcher.InvokeAsync(action).GetAwaiter().GetResult(); 
+#else
                 dispatcher.Invoke(method);
+#endif
```

Comment:
In preview7 there is a dispatcher.Invoke method as well in Avalonia.

Reply:
@maxkatz6 thanks made the change